src/components/app/Details.js

```diff
@@ -79,16 +63,40 @@ class ProfileViewer extends PureComponent<Props> {
     changeSelectedTab(tabSlug);
   };
 
+  _onClickSidebarButton = () => {
+    const { selectedTab, isSidebarOpen, changeSidebarOpenState } = this.props;
+    changeSidebarOpenState(selectedTab, !isSidebarOpen);
+  };
+
   render() {
-    const { tabOrder, changeTabOrder, selectedTab } = this.props;
+    const { tabOrder, selectedTab, isSidebarOpen, changeTabOrder } = this.props;
+    const hasSidebar = selectSidebar(selectedTab) !== null;
+    const extraButton = hasSidebar && (
+      <button
+        className={classNames(
+          'sidebar-open-close-button',
+          'photon-button',
+          'photon-button-ghost',
+          {
+            'sidebar-open-close-button-isopen': isSidebarOpen,
+            'sidebar-open-close-button-isclosed': !isSidebarOpen,
+          }
+        )}
+        title={isSidebarOpen ? 'Close the sidebar' : 'Open the sidebar'}
+        type="button"
+        onClick={this._onClickSidebarButton}
+      />
+    );
+
     return (
       <div className="Details">
         <TabBar
-          tabs={this._tabs}
+          tabs={tabsWithTitleArray}
           selectedTabName={selectedTab}
           tabOrder={tabOrder}
           onSelectTab={this._onSelectTab}
           onChangeTabOrder={changeTabOrder}
+          extraElements={extraButton}
```

Comment:
I went back and forth about how to insert the button in the DOM. In the end I decided to include it in the TabBar, but still define it in `Details`, because I thought that while it was part of the `TabBar` visually the TabBar itself shouldn't have to know anything about the sidebar.